File: Open3D/04_vecindarios_normales/cpp/main.cpp
#include <open3d/Open3D.h>
int main(){ auto pc=open3d::io::CreatePointCloudFromFile("input.ply"); if(!pc) return 1; pc->EstimateNormals(open3d::geometry::KDTreeSearchParamHybrid(0.05,30)); pc->OrientNormalsConsistentTangentPlane(20); open3d::io::WritePointCloud("normals.ply",*pc); }

File: Open3D/04_vecindarios_normales/python/main.py
import open3d as o3d
pc=o3d.io.read_point_cloud("input.ply"); pc.estimate_normals(o3d.geometry.KDTreeSearchParamHybrid(radius=0.05,max_nn=30)); pc.orient_normals_consistent_tangent_plane(20); o3d.io.write_point_cloud("normals.ply",pc)
